cpp/src/io/parquet/writer_impl.cu

```diff
@@ -1409,7 +1465,11 @@ void writer::impl::write(table_view const& table, std::vector<partition_info> co
   }
 
   // Build chunk dictionaries and count pages
-  if (num_chunks != 0) { init_page_sizes(chunks, col_desc, num_columns); }
+  hostdevice_vector<size_type> comp_page_sizes;
+  if (num_chunks != 0) {
+    comp_page_sizes = init_page_sizes(
+      chunks, col_desc, num_columns, max_page_size_bytes, max_page_size_rows, stream);
+  }
```

Comment:
[IILE](https://rigtorp.se/iife/#:~:text=In%20C%2B%2B%20a%20IIFE%20is,of%20statements%20into%20an%20expression.)?
Alternatively, could `init_page_sizes` handle empty input instead?